### ISSUE TITLE
Exclude `abi.encodeX()` calls from `func-named-parameters`

### ISSUE THREAD

**Issue by 0xCLARITY** [protofire/solhint#584]
Solhint regularly complains about the `func-named-parameters` rule when all I'm doing is throwing a bunch of data into `abi.encode()` or `abi.encodePacked()`.

Rather than annotating every single `abi.encodeX()` call with a `//solhint-disable` - I think it makes more sense to exclude `abi.encodeX()` calls from this rule at the rule implementation.

Implementation here: https://github.com/protofire/solhint/pull/583

**Comment by dbale-altoros** [comment]
hello @0xCLARITY thanks a lot for posting and for making a PR
Can you provide an example of what is it failing for you ?
If you are passing more than 3 unnamed parameters to the function... it should complain...
If that is the case, why do you think we need to make an exception with that function call ?
Thanks!

**Comment by dbale-altoros** [comment]
@0xCLARITY I understand what you mean now. 
Thanks a lot for clarification
So in order to make it useful
We should check this at least
[https://docs.soliditylang.org/en/v0.8.26/cheatsheet.html#abi-encoding-and-decoding-functions](https://docs.soliditylang.org/en/v0.8.26/cheatsheet.html#abi-encoding-and-decoding-functions)

And see also if there are others cases similar to this one to avoid this behavior 
I cannot check it now... if you have time, it will be much appreciated

**Comment by 0xCLARITY** [comment]
@dbale-altoros 

I read through the ABI functions and checked their signatures. They are:
```solidity
abi.decode(bytes memory encodedData, (...)) returns (...);
abi.encode(...) returns (bytes memory); 
abi.encodePacked(...) returns (bytes memory);
abi.encodeWithSelector(bytes4 selector, ...) returns (bytes memory);
abi.encodeCall(function functionPointer, (...)) returns (bytes memory);
abi.encodeWithSignature(string memory signature, ...) returns (bytes memory);
```

While there are occasional named parameters (like `abi.encodeWithSelector(selector, ...)`) - every function has `n` positional arguments that I think make them incompatible with the `func-named-parameters` rule.

I also checked the rest of the Solidity built-in functions to see if any of them also had `n` positional arguments - but it appears that only the `abi.*` functions do: https://solang.readthedocs.io/en/latest/language/builtins.html

I think all that is needed is excluding `abi.*` function calls (as currently implemented in #583 )

**Comment by dbale-altoros** [comment]
@0xCLARITY great one
I will review this by the end of the week or monday...
Will include this one in next version with a shootout for you
Please, review the PR because it is failing on the CI 
Whenever passes all the test we can move forward with review!!

Your help is much appreciated!

**Comment by dbale-altoros** [comment]
new release with this feature launched
